### PR TITLE
Feature/an 1877 comment social card v2 fc

### PR DESCRIPTION
**Pull request by paquiquinho**
Changes made to direct the comment flow through the new architecture;

### CODE REVIEW

File: app/src/main/java/cm/aptoide/pt/social/presenter/TimelineNavigator.java [hunk 1]
```diff
@@ -81,9 +79,10 @@ public void navigateToMyAccountView() {
   }
 
   public void navigateToCommentsWithCommentDialogOpen(String cardId) {
-    fragmentNavigator.navigateTo(
-        CommentListFragment.newInstanceWithCommentDialogOpen(CommentType.TIMELINE, cardId,
-            storeContext), true);
+    //fragmentNavigator.navigateTo(
```
Comment (jdandrade) [inline app/src/main/java/cm/aptoide/pt/social/presenter/TimelineNavigator.java]: delete

File: app/src/main/java/cm/aptoide/pt/social/commentslist/PostCommentsPresenter.java [hunk 2]
```diff
@@ -18,19 +19,31 @@
   private final String postId;
   private final Scheduler viewScheduler;
   private final CrashReport crashReporter;
+  private final boolean shouldShow;
```
[the remark this answers is not in the document]
Comment (jdandrade) [reply app/src/main/java/cm/aptoide/pt/social/commentslist/PostCommentsPresenter.java]: also there is a bug associated with this implementation: If user rotates device, a new dialog will always show up, stacking multiple dialogs.

File: app/src/main/java/cm/aptoide/pt/social/commentslist/CommentsNavigator.java [hunk 3]
```diff
@@ -11,21 +13,34 @@
 class CommentsNavigator {
   private final FragmentNavigator fragmentNavigator;
   private final FragmentManager fragmentManager;
+  private final PublishSubject<CommentDataWrapper> subject;
```
[the remark this answers is not in the document]
Comment (paquiquinho) [reply app/src/main/java/cm/aptoide/pt/social/commentslist/CommentsNavigator.java]: fixed

File: app/src/main/java/cm/aptoide/pt/social/commentslist/PostCommentsAdapter.java [hunk 4]
```diff
@@ -85,4 +85,19 @@ void removeLoadMoreProgress() {
     this.comments.remove(progressComment);
     notifyDataSetChanged();
   }
+
+  void showNewComment(Comment comment) {
```
[the remark this answers is not in the document]
Comment (paquiquinho) [reply app/src/main/java/cm/aptoide/pt/social/commentslist/PostCommentsAdapter.java]: fixed

File: app/src/main/java/cm/aptoide/pt/social/commentslist/PostCommentsAdapter.java [hunk 5]
```diff
@@ -85,4 +85,19 @@ void removeLoadMoreProgress() {
...
+    int position = 0;
+    Comment.Parent parent = comment.getParent();
+    if (parent != null) {
+      for (Comment temp : comments) {
+        if (parent.getId() == temp.getId()) {
```
[the remark this answers is not in the document]
Comment (paquiquinho) [reply app/src/main/java/cm/aptoide/pt/social/commentslist/PostCommentsAdapter.java]: fixed

File: app/src/main/java/cm/aptoide/pt/social/commentslist/PostCommentsPresenter.java [hunk 6]
```diff
@@ -18,19 +19,31 @@
...
     this.viewScheduler = viewScheduler;
     this.crashReporter = crashReporter;
     this.postId = postId;
+    this.shouldShow = shouldShow;
   }
 
   @Override public void present() {
+
+    if (shouldShow) {
+      view.getLifecycle()
```
[the remark this answers is not in the document]
Comment (paquiquinho) [reply app/src/main/java/cm/aptoide/pt/social/commentslist/PostCommentsPresenter.java]: fixed

File: app/src/main/java/cm/aptoide/pt/social/presenter/TimelineNavigator.java (hunk 1 above)
Comment (paquiquinho) [reply app/src/main/java/cm/aptoide/pt/social/presenter/TimelineNavigator.java]: fixed

File: app/src/main/java/cm/aptoide/pt/social/commentslist/PostCommentsFragment.java [hunk 7]
```diff
@@ -103,10 +113,14 @@ public static Fragment newInstance(String postId) {
...
-        new CommentsNavigator(getFragmentNavigator(), getActivity().getSupportFragmentManager()),
+        new ArrayList<>()), new CommentMapper(
+        ((AptoideApplication) getContext().getApplicationContext()).getAccountManager())),
+        new CommentsNavigator(getFragmentNavigator(), getActivity().getSupportFragmentManager(),
+            PublishSubject.create()),
         AndroidSchedulers.mainThread(), CrashReport.getInstance(),
-        getArguments().containsKey(POST_ID_KEY) ? getArguments().getString(POST_ID_KEY) : null));
+        getArguments().containsKey(POST_ID_KEY) ? getArguments().getString(POST_ID_KEY) : null,
+        getArguments().containsKey(SHOW_COMMENT_DIALOG) ? getArguments().getBoolean(
+            SHOW_COMMENT_DIALOG) : false));
```
[the remark this answers is not in the document]
Comment (paquiquinho) [reply app/src/main/java/cm/aptoide/pt/social/commentslist/PostCommentsFragment.java]: fixed

File: app/src/main/java/cm/aptoide/pt/social/commentslist/PostCommentsPresenter.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (paquiquinho) [reply app/src/main/java/cm/aptoide/pt/social/commentslist/PostCommentsPresenter.java]: fixed

**Pull request merged**